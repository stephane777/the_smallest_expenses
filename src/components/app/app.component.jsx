import React from "react";

import axios from "axios";
import { Header } from "../header";
import { Footer } from "../footer";
import { Message } from "../resuable/message";
import { Transaction } from "../transaction";

import * as AS from "./app.style";
import { getSmallestExpense, removeIncome } from "../../utils";

const initialState = {
	data: {},
	loading: false,
	error: "",
};

const reducer = (state, action) => {
	switch (action.type) {
		case "FETCHING":
			return {
				...state,
				loading: true,
				error: "",
			};
		case "SUCCESS":
			return {
				...state,
				loading: false,
				data: action.data,
			};
		case "ERROR":
			return {
				...state,
				loading: false,
				error: action.error,
			};
		default:
			return state;
	}
};

const App = () => {
	const [state, dispatch] = React.useReducer(reducer, initialState);
	const { data, loading, error } = state;

	const url = "http://www.mocky.io/v2/5c62e7c33000004a00019b05";

	const formatAmount = (amount) => {
		const formatter = new Intl.NumberFormat("en-GB", {
			style: "currency",
			currency: "GBP",
		});
		return formatter.format(amount);
	};

	React.useEffect(() => {
		const fetchData = async () => {
			try {
				dispatch({ type: "FETCHING" });
				const response = await axios.get(url);
				const result = await response;
				const {
					status,
					data,
					data: { transactions },
				} = await result;
				if (status === 200) {
					const incomeRemoved = removeIncome(transactions);
					const smallestExpense = getSmallestExpense(incomeRemoved);
					const newState = {
						...data,
						transactions: smallestExpense,
					};
					dispatch({ type: "SUCCESS", data: newState });
				} else throw `no data available: ${result.status}`;
			} catch (e) {
				dispatch({ type: "ERROR", error: e.toString() });
			}
		};
		fetchData();
	}, []);

	const expenses = () => {
		const {
			provider: { description, account_number: number, sort_code },
			balance: { amount, currency_iso: currency },
			transactions,
		} = data;

		const formattedAmount = formatAmount(amount);
		return (
			<React.Fragment>
				<AS.FlexColumnAIC className="expense__header">
					<AS.Heading>{description}</AS.Heading>
					<AS.AccountDetails>{`${number} ${sort_code} `}</AS.AccountDetails>
					<AS.Balance>{formattedAmount}</AS.Balance>
				</AS.FlexColumnAIC>
				<AS.FlexRowJCC style={{ marginTop: "4rem" }}>
					<AS.TitleWrapper>
						<AS.Title>Date</AS.Title>
						<AS.Title>Description</AS.Title>
						<AS.Title>Category</AS.Title>
						<AS.Title>Amount</AS.Title>
					</AS.TitleWrapper>
				</AS.FlexRowJCC>
				<AS.FlexRowJCC style={{ marginTop: "1rem" }}>
					<AS.TransactionsWrapper>
						{transactions.map((transaction) => (
							<Transaction key={transaction.id} transaction={transaction} />
						))}
					</AS.TransactionsWrapper>
				</AS.FlexRowJCC>
			</React.Fragment>
		);
	};

	return (
		<div>
			<Header className="header" />
			<AS.AppWrapper>
				{error && <Message message={error} />}
				{loading && !error && <Message message="Loading ..." />}
				{!loading && !error && data.transactions && expenses()}
			</AS.AppWrapper>
			<Footer />
		</div>
	);
};

export { App };
