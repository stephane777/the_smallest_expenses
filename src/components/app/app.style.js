import styled from "styled-components";

export const FlexRowJCC = styled.div`
	display: flex;
	justify-content: center;
`;
export const FlexRowJCS = styled.div`
	display: flex;
	justify-content: start;
`;
export const FlexColumnAIC = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
`;
export const AppWrapper = styled.main`
	min-height: 80vh;
	max-width: 120rem;
	margin: 0 auto;
`;

export const Heading = styled.h1`
	margin: 3rem 0 0.4rem 0;
	font-size: 2.6rem;
	font-weight: 400;
`;

export const AccountDetails = styled.span`
	margin-bottom: 3rem;
	font-size: 1rem;
`;

export const Balance = styled.span`
	font-size: 3rem;
	font-weight: bold;
	color: var(--color-balance);
`;

export const HeadingWrapper = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
`;

export const TitleWrapper = styled(FlexRowJCC)`
	width: 55rem;
	border: 1px solid #ccc;
	border-radius: 6px;
	background-color: #eee;
	box-shadow: 2px 2px 8px #ccc;
	padding: 6px 0 6px 12px;
	@media only screen and (max-width: 768px) {
		width: 45rem;
	}
`;

export const Title = styled(FlexRowJCS)`
	flex: 1;
	font-size: 1.6rem;
`;
export const TransactionsWrapper = styled.ul`
	width: 55rem;
	@media only screen and (max-width: 768px) {
		width: 45rem;
	}
`;
