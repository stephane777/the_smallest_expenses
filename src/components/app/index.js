export { App } from "./app.component.jsx";
