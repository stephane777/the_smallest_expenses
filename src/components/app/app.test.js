import React from "react";
import "@testing-library/jest-dom";
import axios from "axios";
import {
	render,
	screen,
	act,
	cleanup,
	waitForElementToBeRemoved,
} from "@testing-library/react";
import { App } from "./app.component";

import { data } from "./__test__/mockAPI";

jest.mock("axios");

afterEach(cleanup);

describe("it should render App", () => {
	it("should render App", async () => {
		const promise = Promise.resolve(data);
		act(() => {
			const { container } = render(<App />);
			expect(container.firstChild.firstChild.className).toMatch(
				"sc-bdvvaa gbShRg header"
			);
		});
		await act(() => promise);
	});
	it("should show Loading...", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);

		render(<App />);
		expect(screen.getByText(/Loading .../i)).toBeInTheDocument();
		await act(() => promise);
	});
	it("should show Curent Account,sort-code and balance", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);
		render(<App />);

		expect(screen.getByText(/Loading .../i)).toBeInTheDocument();
		await waitForElementToBeRemoved(() => screen.getByText(/Loading .../i));

		expect(screen.getByText(/current account/i)).toBeInTheDocument();
		expect(screen.getByText(/12345678/)).toBeInTheDocument();
		expect(screen.getByText(/12-34-56/)).toBeInTheDocument();
		expect(screen.getByText(/£1,250.32/)).toBeInTheDocument();
		await act(() => promise);
	});
	it("should show the transaction header, Date, Description, Category, Amount", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);
		render(<App />);

		expect(screen.getByText(/Loading .../i)).toBeInTheDocument();
		await waitForElementToBeRemoved(() => screen.getByText(/Loading .../i));

		expect(screen.getByText(/Date/i)).toBeInTheDocument();
		expect(screen.getByText(/Description/i)).toBeInTheDocument();
		expect(screen.getByText(/Category/i)).toBeInTheDocument();
		expect(screen.getByText(/amount/i)).toBeInTheDocument();
		await act(() => promise);
	});
	it("should display the 10 smallest expense", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);
		render(<App />);

		expect(screen.getByText(/Loading .../i)).toBeInTheDocument();
		await waitForElementToBeRemoved(() => screen.getByText(/Loading .../i));

		expect(screen.getByText(/£ -51.19/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -48.75/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -45.23/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -35.87/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -34.98/i)).toBeInTheDocument();

		expect(screen.getByText(/£ -29.99/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -19.99/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -10.76/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -3.75/i)).toBeInTheDocument();
		expect(screen.getByText(/£ -2.85/i)).toBeInTheDocument();

		await act(() => promise);
	});
});
describe("it should match the snapshot", () => {
	it("should match the snapshot while fetching data", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);
		let testedUnit;
		act(() => {
			testedUnit = render(<App />);
		});
		expect(testedUnit).toMatchSnapshot();
		await act(() => promise);
	});
	it("should match the snapshot after fetching data is success", async () => {
		const promise = Promise.resolve(data);
		axios.get.mockImplementation(() => promise);
		let testedUnit;

		act(() => {
			testedUnit = render(<App />);
		});

		await waitForElementToBeRemoved(() => screen.getByText(/Loading .../i));
		expect(testedUnit).toMatchSnapshot();
		await act(() => promise);
	});
});
