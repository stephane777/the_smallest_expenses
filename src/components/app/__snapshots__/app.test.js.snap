// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match the snapshot should match the snapshot after fetching data is success 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdvvaa gbShRg header"
        >
          <img
            class="sc-gsDJrp gLwQTZ"
            data-testid="logo"
            src=""
          />
          <span
            class="sc-dkPtyc fkhCXL"
          >
            monzo
          </span>
        </div>
        <main
          class="sc-egiSv guBZkR"
        >
          
          <div
            class="sc-crHlIS esBBhw expense__header"
          >
            <h1
              class="sc-bqiQRQ eXFeIY"
            >
              Current Account
            </h1>
            <span
              class="sc-ksdxAp kTfBcU"
            >
              12345678 12-34-56 
            </span>
            <span
              class="sc-hBURRC eTXnbk"
            >
              £1,250.32
            </span>
          </div>
          <div
            class="sc-kDThTU imuaVC"
            style="margin-top: 4rem;"
          >
            <div
              class="sc-kDThTU sc-fFehDp imuaVC epIRLE"
            >
              <div
                class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
              >
                Date
              </div>
              <div
                class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
              >
                Description
              </div>
              <div
                class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
              >
                Category
              </div>
              <div
                class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
              >
                Amount
              </div>
            </div>
          </div>
          <div
            class="sc-kDThTU imuaVC"
            style="margin-top: 1rem;"
          >
            <ul
              class="sc-ieebsP qgUck"
            >
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-01
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Asda
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Shopping
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -51.19
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-06-29
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  British Gas
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Bills
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -48.75
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-01
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Marks & Spencer
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Shopping
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -45.23
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-09
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Pret a Manger
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Eating Out
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -35.87
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-03
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Tesco
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Groceries
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -34.98
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-02
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Vodafone
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Entertainment
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -29.99
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-03
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Pact Coffee
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Shopping
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -19.99
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-01
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Wilko
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Shopping
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -10.76
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-04
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Costa Coffee
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Eating Out
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -3.75
                </span>
              </li>
              <li
                class="sc-furvIG krqrmL transaction"
              >
                <span
                  class="sc-pVTma ddErkt"
                >
                  2018-07-05
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Transport for London
                </span>
                <span
                  class="sc-pVTma ddErkt"
                >
                  Transport
                </span>
                <span
                  class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
                >
                  £ -2.85
                </span>
              </li>
            </ul>
          </div>
        </main>
        <div
          class="sc-hKwCoD CQaOl footer"
        >
          <footer
            class="sc-eCImvq PvPWG"
          >
            <a
              class="footer__link"
              data-testid="github-link"
              href="https://github.com/stephane777"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="sc-gKckTs lehCFg"
                data-testid="github-svg"
              >
                <use
                  class="grid__icon-use"
                  href="#icon-github"
                />
              </svg>
            </a>
            <p
              class="sc-jRQAMF fvfIEH"
            >
              Created by Stephane Candelas.
            </p>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdvvaa gbShRg header"
      >
        <img
          class="sc-gsDJrp gLwQTZ"
          data-testid="logo"
          src=""
        />
        <span
          class="sc-dkPtyc fkhCXL"
        >
          monzo
        </span>
      </div>
      <main
        class="sc-egiSv guBZkR"
      >
        
        <div
          class="sc-crHlIS esBBhw expense__header"
        >
          <h1
            class="sc-bqiQRQ eXFeIY"
          >
            Current Account
          </h1>
          <span
            class="sc-ksdxAp kTfBcU"
          >
            12345678 12-34-56 
          </span>
          <span
            class="sc-hBURRC eTXnbk"
          >
            £1,250.32
          </span>
        </div>
        <div
          class="sc-kDThTU imuaVC"
          style="margin-top: 4rem;"
        >
          <div
            class="sc-kDThTU sc-fFehDp imuaVC epIRLE"
          >
            <div
              class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
            >
              Date
            </div>
            <div
              class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
            >
              Description
            </div>
            <div
              class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
            >
              Category
            </div>
            <div
              class="sc-iqsfdx sc-bkkfTU bOVjmG enzepX"
            >
              Amount
            </div>
          </div>
        </div>
        <div
          class="sc-kDThTU imuaVC"
          style="margin-top: 1rem;"
        >
          <ul
            class="sc-ieebsP qgUck"
          >
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-01
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Asda
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Shopping
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -51.19
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-06-29
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                British Gas
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Bills
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -48.75
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-01
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Marks & Spencer
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Shopping
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -45.23
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-09
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Pret a Manger
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Eating Out
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -35.87
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-03
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Tesco
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Groceries
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -34.98
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-02
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Vodafone
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Entertainment
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -29.99
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-03
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Pact Coffee
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Shopping
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -19.99
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-01
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Wilko
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Shopping
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -10.76
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-04
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Costa Coffee
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Eating Out
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -3.75
              </span>
            </li>
            <li
              class="sc-furvIG krqrmL transaction"
            >
              <span
                class="sc-pVTma ddErkt"
              >
                2018-07-05
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Transport for London
              </span>
              <span
                class="sc-pVTma ddErkt"
              >
                Transport
              </span>
              <span
                class="sc-pVTma sc-jrQzUz ddErkt iKvSKb"
              >
                £ -2.85
              </span>
            </li>
          </ul>
        </div>
      </main>
      <div
        class="sc-hKwCoD CQaOl footer"
      >
        <footer
          class="sc-eCImvq PvPWG"
        >
          <a
            class="footer__link"
            data-testid="github-link"
            href="https://github.com/stephane777"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="sc-gKckTs lehCFg"
              data-testid="github-svg"
            >
              <use
                class="grid__icon-use"
                href="#icon-github"
              />
            </svg>
          </a>
          <p
            class="sc-jRQAMF fvfIEH"
          >
            Created by Stephane Candelas.
          </p>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`it should match the snapshot should match the snapshot while fetching data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="sc-bdvvaa gbShRg header"
        >
          <img
            class="sc-gsDJrp gLwQTZ"
            data-testid="logo"
            src=""
          />
          <span
            class="sc-dkPtyc fkhCXL"
          >
            monzo
          </span>
        </div>
        <main
          class="sc-egiSv guBZkR"
        >
          
          <div
            class="sc-iCfLBT bzGHbE message"
          >
            <span>
              Loading ...
            </span>
          </div>
        </main>
        <div
          class="sc-hKwCoD CQaOl footer"
        >
          <footer
            class="sc-eCImvq PvPWG"
          >
            <a
              class="footer__link"
              data-testid="github-link"
              href="https://github.com/stephane777"
              rel="noreferrer"
              target="_blank"
            >
              <svg
                class="sc-gKckTs lehCFg"
                data-testid="github-svg"
              >
                <use
                  class="grid__icon-use"
                  href="#icon-github"
                />
              </svg>
            </a>
            <p
              class="sc-jRQAMF fvfIEH"
            >
              Created by Stephane Candelas.
            </p>
          </footer>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="sc-bdvvaa gbShRg header"
      >
        <img
          class="sc-gsDJrp gLwQTZ"
          data-testid="logo"
          src=""
        />
        <span
          class="sc-dkPtyc fkhCXL"
        >
          monzo
        </span>
      </div>
      <main
        class="sc-egiSv guBZkR"
      >
        
        <div
          class="sc-iCfLBT bzGHbE message"
        >
          <span>
            Loading ...
          </span>
        </div>
      </main>
      <div
        class="sc-hKwCoD CQaOl footer"
      >
        <footer
          class="sc-eCImvq PvPWG"
        >
          <a
            class="footer__link"
            data-testid="github-link"
            href="https://github.com/stephane777"
            rel="noreferrer"
            target="_blank"
          >
            <svg
              class="sc-gKckTs lehCFg"
              data-testid="github-svg"
            >
              <use
                class="grid__icon-use"
                href="#icon-github"
              />
            </svg>
          </a>
          <p
            class="sc-jRQAMF fvfIEH"
          >
            Created by Stephane Candelas.
          </p>
        </footer>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
